internal/services/watchlist/watchlist.go

```diff
@@ -0,0 +1,89 @@
+package watchlist
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"errors"
+)
+
+func CreateWatchlist(Req CreateRequest) (Res CreateResponse, err error) {
+	var user model.Users
+	err = config.DB.Table("users").Where("user_id = ?", Req.UserId).First(&user).Error
+	if err != nil {
+		return Res, errors.New("user not found")
+	}
+	var wl model.Watchlist
+	err = config.DB.Table("watchlist").Where("name = ?", Req.WatchlistName).First(&wl).Error
+	if err == nil {
+		return Res, errors.New("watchlist name already exists")
+	}
+	wl.Name = Req.WatchlistName
+	wl.Userid = Req.UserId
+	err = config.DB.Table("watchlist").Create(&wl).Error
+	if err != nil {
```

Comment:
why aren't we creating watchlist for user here?

Reply:
that will be created when the user enters a stock in the watchlist